src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -344,8 +346,29 @@ public object VisitTable(TableExpression tableExpression, SearchOptions context)
                     var includeExpression = (IncludeExpression)tableExpression.NormalizedPredicate;
 
                     StringBuilder.Append("SELECT DISTINCT ");
-                    StringBuilder.Append(VLatest.Resource.ResourceSurrogateId, referenceTargetResourceTableAlias).AppendLine(" AS Sid1, 0 AS IsMatch")
-                        .Append("FROM ").Append(VLatest.ReferenceSearchParam).Append(' ').AppendLine(referenceSourceTableAlias)
+
+                    if (includeExpression.Reversed)
+                    {
+                        // In case its revinclude, we limit the number of returned items as the resultset size is potentially
+                        // unbounded.
+                        StringBuilder.Append(" TOP (").Append(Parameters.AddParameter(MaxIncludedItems)).Append(") ");
+                    }
+                    else
+                    {
+                        StringBuilder.Append("0 AS IsPartial, "); // can't be partial as we are requesting all includes
```

Comment:
Is it possible to keep the same order of the SELECT statement, meaning Sid, IsMatch, IsPartial?

Reply:
done